openquake/hazardlib/gsim/zhao_2016_volc_perg/volc_perg.py

```diff
@@ -0,0 +1,436 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2012-2023 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import fiona
+import pandas as pd
+
+from collections import OrderedDict
+from openquake.hazardlib.geo import Point
+from openquake.hazardlib.geo import polygon as pgn
+from openquake.hazardlib.geo.geodetic import npoints_between, distance
+from openquake.hazardlib.geo.surface import PlanarSurface
+#from openquake.hazardlib.source.rupture import BaseRupture
+from openquake.hazardlib.scalerel.wc1994 import WC1994
+
+
+def get_volc_zones(volc_fname):
+    """
+    Read in volcanic zones from GeoJSON to construct polygons of each zone
+    :param volc_fname:
+        volc_fname: Path to GeoJSON with polygon for each volcanic zone and
+        zone id
+    """
+    # Get the volc zone polygons
+    with fiona.open(volc_fname,'r') as inp:
```

Comment:
This usage is absolutely FORBIDDEN. You cannot read files from a GMPE without restrictions as you are doing here. The right way to implement GMPEs depending on file parameters is documented in the manual: https://docs.openquake.org/oq-engine/advanced/master/classical_PSHA.html#file-dependent-gmpes